.github/workflows/ci.yml

```diff
@@ -17,272 +17,275 @@
 #
 
 ---
-name: Airflow
+name: CI Build
 on:
+  schedule:
+    - cron: '10 2 * * *'
   push:
-    branches: ['*']
+    branches: ['master', 'v1-10-test', 'v1-10-stable']
   pull_request:
-    branches: [master]
+    branches: ['master', 'v1-10-test', 'v1-10-stable']
 env:
-  BUILD_ID: ${{github.sha }}
   MOUNT_LOCAL_SOURCES: "false"
-  MOUNT_HOST_AIRFLOW_VOLUME: "true"
   FORCE_ANSWER_TO_QUESTIONS: "yes"
   SKIP_CHECK_REMOTE_IMAGE: "true"
+  SKIP_CI_IMAGE_CHECK: "true"
   DB_RESET: "true"
   VERBOSE: "true"
-  CI: "true"
-  # Should be a target branch
-  TRAVIS_BRANCH: "master"
-  TRAVIS: "true"
+  UPGRADE_TO_LATEST_REQUIREMENTS: "false"
+  ENABLED_INTEGRATIONS: "cassandra kerberos mongo openldap presto rabbitmq redis"
+  PYTHON_MAJOR_MINOR_VERSION: 3.6
+  # TODO: Enable cache registry once a master build succeeds uploading the images to Github registry cache
+  DISABLE_CACHE_REGISTRY: "true"
+  CACHE_REGISTRY: "docker.pkg.github.com"
+  CACHE_IMAGE_PREFIX: ${{ github.repository }}
+  CACHE_REGISTRY_USERNAME: ${{ github.actor }}
+  CACHE_REGISTRY_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+
 jobs:
-  statics:
-    name: Static checks
+
+  static-checks-1:
+    name: "Checks: pylint, mypy"
     runs-on: ubuntu-latest
     env:
-      PYTHON_VERSION: 3.6
-      AIRFLOW_MOUNT_SOURCE_DIR_FOR_STATIC_CHECKS: "true"
-      TRAVIS_JOB_NAME: "Static"
+      MOUNT_SOURCE_DIR_FOR_STATIC_CHECKS: "true"
+      CI_JOB_TYPE: "Pylint checks for tests"
     steps:
       - uses: actions/checkout@master
-      - name: free disk space
-        run: |
-          sudo swapoff -a
-          sudo rm -f /swapfile
-          sudo apt clean
-          docker rmi $(docker image ls -aq)
-          df -h
       - uses: actions/setup-python@v1
         with:
-          python-version: ${{ env.PYTHON_VERSION }}
-      - name: "Before install"
-        run: ./scripts/ci/ci_before_install.sh
-      - name: "Static checks"
-        if: success()
+          python-version: '3.x'
+      - name: Cache pre-commit env
+        uses: actions/cache@v1
         env:
-          PYTHON_VERSION: 3.6
-        run: ./scripts/ci/ci_run_all_static_checks.sh
+          cache-name: cache-pre-commit
+        with:
+          path: ~/.cache/pre-commit
```

Comment:
Can you explain the purpose of this cache?

Reply:
Yep. it speeds up (by about a minute) pre-commit initialization. When you run pre-commit first time it builds local virtualenvs for all the pre-commits requiring it. It takes about a minute and the .venvs are stored in ~/.cache/pre-commit . This way we save about a minute for static checks.